src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -210,15 +233,172 @@ public class DeploymentOptions
                 ObjectType.DatabaseOptions,
                 ObjectType.EventNotifications,
                 ObjectType.ServerRoleMembership,
-                ObjectType.AssemblyFiles,
-        };
+                ObjectType.AssemblyFiles
+            }
+        );
+
+        public DeploymentOptionProperty<bool> AllowExternalLibraryPaths { get; set; }
+
+        public DeploymentOptionProperty<bool> AllowExternalLanguagePaths { get; set; }
+
+        public DeploymentOptionProperty<bool> DoNotEvaluateSqlCmdVariables { get; set; }
+
+        public DeploymentOptionProperty<bool> DisableParallelismForEnablingIndexes { get; set; }
+
+        public DeploymentOptionProperty<bool> DoNotDropWorkloadClassifiers { get; set; }
+
+        public DeploymentOptionProperty<bool> DisableIndexesForDataPhase { get; set; }
+
+        public DeploymentOptionProperty<bool> DoNotDropDatabaseWorkloadGroups { get; set; }
+
+        public DeploymentOptionProperty<bool> HashObjectNamesInLogs { get; set; }
+
+        public DeploymentOptionProperty<bool> IgnoreWorkloadClassifiers { get; set; }
+
+        public DeploymentOptionProperty<bool> IgnoreDatabaseWorkloadGroups { get; set; }
+
+        public DeploymentOptionProperty<bool> IsAlwaysEncryptedParameterizationEnabled { get; set; }
+
+        public DeploymentOptionProperty<bool> PreserveIdentityLastValues { get; set; }
+
+        public DeploymentOptionProperty<bool> RestoreSequenceCurrentValue { get; set; }
+
+        public DeploymentOptionProperty<bool> RebuildIndexesOfflineForDataPhase { get; set; }
+
+        private Dictionary<string, string> _displayNameMapDict;
 
         #endregion
 
+        /// <summary>
+        /// Mapping the DisplayName to the dac deploy option
+        /// Adding new properties here would give easy handling of new option to all extensions
+        /// </summary>
+        private void SetDisplayNameForOption()
+        {
+            #region Display Name and Dac Options Mapping
+            DacDeployOptions d = new DacDeployOptions();
+            _displayNameMapDict = new Dictionary<string, string>();
+
+            // Ex: displayNameMapDict["DacFx Option Name"] = "ADS UI Display Name"
+            _displayNameMapDict[nameof(d.AdditionalDeploymentContributorArguments)] = "Additional Deployment Contributor Arguments";
```

Comment:
Is this stuff something you could put into DacFX instead? Having to define it here still isn't great since that means every consumer of this functionality has to redefine these strings. @kisantia FYI

Reply:
@udeeshagautam this is what we discussed to get the display names from the DacFx like Description that we are already getting, which can be reusable at all extensions. I'll create a new task for this one and will be working on right after this PR.